ivy/func_wrapper.py

```diff
@@ -771,6 +772,22 @@ def _infer_device(*args, device=None, **kwargs):
     return _infer_device
 
 
+def auto_shift_device(fn: Callable) -> Callable:
+    def _auto_shift_device(*args, **kwargs):
+        if ivy.functional.device.soft_device_placement and ivy.get_array_mode():
```

Comment:
we should be making use of something like `ivy.get_soft_device_mode` instead of `ivy.functional.device.soft_device_placement`